File: T7/codigos2021/T07_API/app/src/main/java/com/example/t07_api/DialogoPregunta.java
package com.example.t07_api;

import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.Button;
import android.widget.ImageButton;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.DialogFragment;

public class DialogoPregunta extends DialogFragment {

    private View view;
    private ImageButton botonOk, botonFail;
    private OnDialogoListener listener;

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        listener = (OnDialogoListener) context;
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {

        AlertDialog.Builder builder = new AlertDialog.Builder(getContext());
        view = LayoutInflater.from(getContext()).inflate(R.layout.layout_dialogo,null,false);
        builder.setView(view);
        return builder.create();
    }

    @Override
    public void onStart() {
        super.onStart();
        botonFail = view.findViewById(R.id.boton_incorrecto);
        botonOk = view.findViewById(R.id.boton_correcto);
        botonOk.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                listener.onDialogoSelected(true);
                dismiss();
            }
        });
        botonFail.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                listener.onDialogoSelected(false);
                dismiss();
            }
        });
    }

    public interface OnDialogoListener{
        void onDialogoSelected(boolean respuesta);
    }

}

File: Kotlin/T5/codigos_ces/Repaso/app/src/main/java/com/example/repaso/ui/dialog/DialogoPregunta.kt
package com.example.repaso.ui.dialog

import android.app.Dialog
import android.content.Context
import android.content.DialogInterface
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
import com.example.repaso.adapter.AdapterTrivial
import com.example.repaso.model.Pregunta

class DialogoPregunta : DialogFragment() {

    private lateinit var pregunta: Pregunta
    private var seleccion: String? = null
    private lateinit var listener: OnRespuestaListener

    companion object {
        // variables
        // metodos -> crear una instancia del objeto con parametros
        ;
        fun newInstance(pregunta: Pregunta): DialogoPregunta {
            val dialogoPregunta: DialogoPregunta = DialogoPregunta()
            val bundle: Bundle = Bundle()
            bundle.putSerializable("pregunta", pregunta)
            dialogoPregunta.arguments = bundle
            return dialogoPregunta
        }
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        this.pregunta = this.arguments?.getSerializable("pregunta") as Pregunta
        this.listener = context as OnRespuestaListener
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder: AlertDialog.Builder = AlertDialog.Builder(requireActivity())
        builder.setTitle(pregunta.question)
        (pregunta.incorrectAnswers as ArrayList<String>)
            .add(pregunta.correctAnswer!!)
        val opciones: Array<CharSequence> = pregunta.incorrectAnswers
                as Array<CharSequence>
        opciones.shuffle();
        builder.setSingleChoiceItems(opciones, -1, { _, i ->
            seleccion = opciones[i].toString()
        })
        builder.setPositiveButton("Seleccionar", { _, _ ->
            // pasar a la activity la opcion seleccionada
            if (seleccion != null) {
                listener.onRespuestaSelected(seleccion!!)
            }
        })
        builder.setNeutralButton("Cancelar", { _, _ -> })

        return builder.create()
    }

    interface OnRespuestaListener {
        fun onRespuestaSelected(opcion: String)
    }
}